tests-web/www/js/main.js

```diff
@@ -1,7 +1,230 @@
+  }
+
+  removeLayer(layer) {
+    this.map.removeLayer(layer);
+    this.layerControl.removeLayer(layer);
+  }
+
+  async setCurrentTest(testMaker) {
+    this.currentTest?.cleanup();
+    this.currentTest = await testMaker(this);
+    this.map.fitBounds(this.currentTest.bounds, { animate: false });
+    this.currentTest.renderControls(document.getElementById("view-controls"));
+  }
+}
+
+// design litmus test: can multiple of these exist at the app at the same time?
+class TestCase {
+  // null for freshly imported places
+  constructor(app, name, osmXML, drivingSide, layers, bounds) {
+    this.app = app;
+    this.name = name;
+    this.osmXML = osmXML;
+    this.drivingSide = drivingSide;
+    // TODO this is probably more organized at some point, not just a list
+    this.layers = layers;
+    this.bounds = bounds;
+  }
+
+  cleanup() {
+    for (const layer of this.layers) {
+      this.app.removeLayer(layer);
+    }
+  }
+
+  static async loadFromServer(app, name) {
+    const prefix = `tests/${name}/`;
+    const osmInput = await loadFile(prefix + "input.osm");
+    const rawMap = loadFile(prefix + "raw_map.json");
+    const network = loadFile(prefix + "road_network.dot");
+
+    const rawMapLayer = makePlainGeoJsonLayer(await rawMap);
+    const bounds = rawMapLayer.getBounds();
+
+    var layers = [];
+    layers.push(app.addLayer("Geometry", rawMapLayer));
+    layers.push(app.addLayer("OSM", makeOsmLayer(osmInput)));
+    layers.push(
+      app.addLayer("Network", await makeDotLayer(await network, { bounds }))
+    );
+
+    const drivingSide = JSON.parse(await loadFile(prefix + "test.json"))[
+      "driving_side"
+    ];
+
+    return new TestCase(app, name, osmInput, drivingSide, layers, bounds);
+  }
+
+  static async importCurrentView(app, importButton) {
+    // Grab OSM XML from Overpass
+    // (Sadly toBBoxString doesn't seem to match the order for Overpass)
+    const b = app.map.getBounds();
+    const bbox = `${b.getSouth()},${b.getWest()},${b.getNorth()},${b.getEast()}`;
+    const query = `(nwr(${bbox}); node(w)->.x; <;); out meta;`;
+    const url = `https://overpass-api.de/api/interpreter?data=${query}`;
+    console.log(`Fetching from overpass: ${url}`);
+
+    importButton.innerText = "Downloading from Overpass...";
+    // Prevent this function from happening twice in a row. It could also maybe
+    // be nice to allow cancellation.
+    importButton.disabled = true;
+
+    // TODO If this fails, I guess app.currentTest winds up null?
+    try {
+      const resp = await fetch(url);
+      const osmInput = await resp.text();
+
+      importButton.innerText = "Importing OSM data...";
+
+      // TODO Ask overpass
+      const drivingSide = "Right";
+
+      const network = new JsStreetNetwork(osmInput, {
+        driving_side: drivingSide,
+      });
+      const rawMapLayer = makePlainGeoJsonLayer(network.toGeojsonPlain());
+      const bounds = rawMapLayer.getBounds();
+
+      var layers = [];
```

Comment:
There's some duplicate code with the reimport case below. Maybe we could have a helper that fills out layers given a `JsStreetNetwork`? But there are some differences.. we want to name the layers with "(reimport)" in one case. And in this case, we don't have the `TestCase` created yet, but we do in the reimport case